main_test.go

```diff
@@ -21,294 +21,183 @@
-	mockOSWrapper := new(mockOSWrapper)
-	pprofer := defaultPprofer{
-		osWrapper: mockOSWrapper,
+func getTempFilename(t *testing.T, suffix string) string {
+	f, err := ioutil.TempFile("", "")
+	if err != nil {
+		t.Fatalf("Failed to create temp file: %v", err)
 	}
 
-	mockOSWrapper.On("cmdOutput", mock.AnythingOfType("*exec.Cmd")).Return(nil, errors.New("pprof underlying error")).Once()
-
-	sampleArgs := []string{"-seconds", "15", "http://localhost:8080"}
-	out, err := pprofer.runPprofCommand(sampleArgs...)
-
-	assert.Equal(t, 0, len(out))
-	assert.Error(t, err)
-	mockOSWrapper.AssertExpectations(t)
+	defer f.Close()
+	return f.Name() + suffix
+}
+
+func TestRunFile(t *testing.T) {
+	opts := getDefaultOptions()
+	opts.File = getTempFilename(t, ".svg")
+
+	withScriptsInPath(t, func() {
+		if err := runWithOptions(opts); err != nil {
+			t.Fatalf("Run with Print failed: %v", err)
+		}
+
+		f, err := os.Open(opts.File)
+		if err != nil {
+			t.Errorf("Failed to open output file: %v", err)
+		}
+		defer f.Close()
+
+		// Our fake flamegraph scripts just add script names to the output.
+		reader := bufio.NewReader(f)
+		line1, err := reader.ReadString('\n')
+		if err != nil {
+			t.Errorf("Failed to read line 1 in output file: %v", err)
+		}
+		line2, err := reader.ReadString('\n')
+		if err != nil {
+			t.Errorf("Failed to read line 2 in output file: %v", err)
+		}
+
+		if !strings.Contains(line1, "flamegraph.pl") ||
+			!strings.Contains(line2, "stackcollapse.pl") {
+			t.Errorf("Output file has not been processed by flame graph scripts")
+		}
+	})
 }
 
-// 'go tool pprof' doesn't exit on errors with nonzero status codes. This test
-// ensures that go-torch will detect undrlying errors despite the pprof bug.
-// See pprof issue here https://github.com/golang/go/issues/11510
-func TestRunPprofCommandHandlePprofErrorBug(t *testing.T) {
-	mockOSWrapper := new(mockOSWrapper)
-	pprofer := defaultPprofer{
-		osWrapper: mockOSWrapper,
-	}
-
-	mockOSWrapper.On("cmdOutput", mock.AnythingOfType("*exec.Cmd")).Return([]byte{}, nil).Once()
-
-	sampleArgs := []string{"-seconds", "15", "http://localhost:8080"}
-	out, err := pprofer.runPprofCommand(sampleArgs...)
+func TestRunBadFile(t *testing.T) {
+	opts := getDefaultOptions()
+	opts.File = "/dev/zero/invalid/file"
 
-	assert.Equal(t, 0, len(out))
-	assert.Error(t, err)
-	mockOSWrapper.AssertExpectations(t)
+	withScriptsInPath(t, func() {
+		if err := runWithOptions(opts); err == nil {
+			t.Fatalf("Run with bad file expected to fail")
+		}
+	})
 }
 
-func TestNewTorcher(t *testing.T) {
-	assert.NotNil(t, newTorcher())
-}
+func TestRunPrint(t *testing.T) {
+	opts := getDefaultOptions()
+	opts.Print = true
 
-func TestNewCommander(t *testing.T) {
-	assert.NotNil(t, newCommander())
+	withScriptsInPath(t, func() {
+		if err := runWithOptions(opts); err != nil {
+			t.Fatalf("Run with Print failed: %v", err)
+		}
+		// TODO(prashantv): Verify that output is printed to stdout.
```

Comment:
future TODO - it becomes a lot cleaner with some future changes I want to make, but that involves changing how `--raw` works, so I want to do that separately.